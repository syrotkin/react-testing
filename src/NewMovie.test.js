import React from 'react';
import { render, fireEvent, cleanup } from 'react-testing-library';
import NewMovie from './NewMovie';

afterEach(cleanup);

test('<NewMovie />', () => {
  const {
    getByTestId, debug, queryByTestId, container
  } = render(<NewMovie />);

  // console.log(container.firstChild); // this is HTMLDivElement
  const element = queryByTestId('page-title');
  if (element) {
    expect(getByTestId('page-title').textContent).toBe('New Movie');
  }

  // this works because it is deep rendering, NewMovie includes MovieForm
  expect(queryByTestId('movie-form')).toBeTruthy();
  expect(container.firstChild).toMatchSnapshot();
});
