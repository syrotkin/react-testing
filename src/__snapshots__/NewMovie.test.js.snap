// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewMovie /> 1`] = `
<div>
  <h2
    data-testid="page-title"
  >
    New Movie
  </h2>
  <div>
    <form
      data-testid="movie-form"
    >
      <input
        type="text"
      />
      <button
        type="button"
      >
        Submit
      </button>
    </form>
  </div>
</div>
`;
